### PR TITLE
Feature/myzimbra x

### PR DESCRIPTION
**Pull request by girishbhamare**
I have created "feature/myzimbraX" branch  from "feature/zimbraX".
Rebase with develop.
Added following files.

- ListOfTestCases_zimbraX.txt: It has files which we need to skip while running in ZimbraX set up
- global.properties.zimbrax: Server and other details specific to zimbraX env.
- src/java/com/zimbra/qa/soap/SoapTest.java: Adding polling interval for "Get"
--Added ZimbraX tag specific to zimbraX tests.

Executed run on ZimbraX as well as zimbra 8.
We have got good result.
Zimbra 8 Result:

Executed:4561
Pass:4518
Fail:43
Script Errors:10

Most of the failure are passed on rerun

Following tests we need to skip while running in Zimbra8

	/opt/qa/soapvalidator/data/soapvalidator/Admin/Accounts/account_create.xml
	/opt/qa/soapvalidator/data/soapvalidator/Admin/Accounts/account_getinfo.xml
	/opt/qa/soapvalidator/data/soapvalidator/Admin/Accounts/account_modify.xml
      /opt/qa/soapvalidator/data/soapvalidator/MailClient/ClientInfoRequest.xml

### CODE REVIEW

File: data/soapvalidator/MailClient/Search/Bugs/bug72260.xml [hunk 1]
```diff
@@ -85,7 +85,7 @@
 		<t:test>
 			<t:request>
 				<SearchRequest xmlns="urn:zimbraMail" types="message">
-					<query>"to:kevinh"</query>
+					<query>to:"kevinh"</query>
```
Comment (swapnilpingle89) [inline data/soapvalidator/MailClient/Search/Bugs/bug72260.xml]: Does this work on legacy and zimbraX both?

File: src/java/com/zimbra/qa/soap/RestServletTest.java [hunk 2]
```diff
@@ -360,7 +360,7 @@ private boolean containsText(String contentType) {
 	}
 
 	private boolean isBinary(String contentType) {
-		if (contentType.equalsIgnoreCase("application/zip")) {
+		if (contentType.equalsIgnoreCase("application/zip") || contentType.equalsIgnoreCase("application/x-compressed-tar")) {
```
Comment (swapnilpingle89) [inline src/java/com/zimbra/qa/soap/RestServletTest.java]: Does this affect legacy rest tests in any way?

File: src/java/com/zimbra/qa/soap/SoapTestMain.java [hunk 3]
```diff
@@ -213,7 +213,7 @@ public static void parseArgs(String args[]) throws HarnessException {
         		{
         			String type = iterator.next();
         			
-        			if (type.equalsIgnoreCase("sanity")) {
+        			if (type.equalsIgnoreCase("sanity") && !globalProperties.getProperty("ignoreFiles").toString().equalsIgnoreCase("true")) {
```
Comment (swapnilpingle89) [inline src/java/com/zimbra/qa/soap/SoapTestMain.java]: Lets not change this behavior in legacy.

File: build.xml [hunk 4]
```diff
@@ -517,11 +518,11 @@
   	<target name="build-testware" description=" Build soap harness testware." 
   								depends="clean,build-testware-one,build-testware-two,build-tools-setup,compress-move" />		
 
-		<target name="Run-SoapTestCore" depends="compile" description="Run Staf Tests in non-STAF environment">
-			<property name="testRoot" value="data/soapvalidator/MailClient/Auth/auth_basic.xml"/>
+		<target name="Run-SoapTestCore" depends="compile,zm-network-soap-content" description="Run Staf Tests in non-STAF environment">
```
Comment (swapnilpingle89) [inline build.xml]: Please create different ant target for zimbraX instead of changing existing one.

File: build.xml [hunk 5]
```diff
@@ -423,12 +423,13 @@
 		<target name="zm-network-soap-content" description="Copies network related soap data and tests">
 			<mkdir dir="${build.temp.dir}"/>
     		<echo message="Copying network related data and tests..."/>
-			<if><available file="${zm-network-soap-harness.dir}/conf/setup.sh" type="file"/>
+			<!--Current set of network related changes do not need these steps to be executed in zimbraX. Hence this part has been commented.-->
```
Comment (swapnilpingle89) [inline build.xml]: This should be uncommented

File: conf/global.properties [hunk 6]
```diff
@@ -143,6 +144,9 @@ postfixrequests.list=mail:SendMsgRequest,\
                                         mail:BounceMsgRequest,\
                                         mail:ForwardAppointmentInviteRequest,\
 										admin:ContactBackupRequest
+ewsrequests.list=ewsmsg:SyncFolderItems
+ewsrequests.delay = 4000
+EwsDelay.check = true
```
Comment (swapnilpingle89) [inline conf/global.properties]: This should be false by default. As we don't want this delay in legacy

**Comment by swapnilpingle89** [comment]
Please squash all the commits in one commit.

File: build.xml (hunk 5 above)
Comment (girishbhamare) [reply build.xml]: Done

File: build.xml (hunk 4 above)
Comment (girishbhamare) [reply build.xml]: done

File: conf/ListOfTestCases.txt [hunk 7]
```diff
@@ -1,4 +1,87 @@
-data/soapvalidator/Admin/Domains/domains_get.xml
-data/soapvalidator/Admin/AdminSearch/Aliases
-data/soapvalidator/MailClient/Item
-data/soapvalidator/MailClient/Mail/message_action.xml
\ No newline at end of file
+data/soapvalidator/Admin/ACL/acl_basic.xml
```
[the remark this answers is not in the document]
Comment (girishbhamare) [reply conf/ListOfTestCases.txt]: done

File: conf/global.properties [hunk 8]
```diff
@@ -110,7 +111,7 @@ iCalDomain.name=testdomain.com
 soapxml.root=data/soapvalidator
 zdesktopsoapxml.root=data/zdesktopvalidator
 testMailRaw.root=data/TestMailRaw
-network.testMailRaw.root=build/temp/data/TestMailRaw
+testMailRaw.network.root=build/temp/data/TestMailRaw
```
[the remark this answers is not in the document]
Comment (girishbhamare) [reply conf/global.properties]: Verified this  change "testMailRaw.network.root" in legacy as well as zimbra X and its working fine.

File: data/soapvalidator/Admin/Accounts/account_getinfo.xml [hunk 9]
```diff
@@ -106,7 +109,19 @@
             <t:select path='//admin:CreateAccountResponse/admin:account/admin:a[@n="zimbraMailHost"]' set="test_acct4.server"/>
         </t:response>
     </t:test>
-
+    <t:test required="true">
```
[the remark this answers is not in the document]
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/account_getinfo.xml]: required = true  is everywhere in this file, its just creating extra account.

File: data/soapvalidator/Admin/Accounts/account_create.xml [hunk 10]
```diff
@@ -9495,7 +9495,25 @@ ZimbraMailDelivery is no longer used
     </t:test>  
     
 </t:test_case>
-    
+
```
[the remark this answers is not in the document]
Comment (girishbhamare) [reply data/soapvalidator/Admin/Accounts/account_create.xml]: Added "ZimbraX" wherever needed.

File: data/soapvalidator/MailClient/Search/Bugs/bug72260.xml (hunk 1 above)
Comment (girishbhamare) [reply data/soapvalidator/MailClient/Search/Bugs/bug72260.xml]: yes , this is working fine in zimbrax and Legacy

File: src/java/com/zimbra/qa/soap/RestServletTest.java (hunk 2 above)
Comment (girishbhamare) [reply src/java/com/zimbra/qa/soap/RestServletTest.java]: nope

File: src/java/com/zimbra/qa/soap/SoapTestMain.java (hunk 3 above)
Comment (girishbhamare) [reply src/java/com/zimbra/qa/soap/SoapTestMain.java]: It doesn't affect Legacy

**Comment by swapnilpingle89** [comment]
Please squash all the commits before merging

File: conf/ListOfTestCases.txt [hunk 11]
```diff
@@ -1,4 +1,4 @@
 data/soapvalidator/Admin/Domains/domains_get.xml
 data/soapvalidator/Admin/AdminSearch/Aliases
 data/soapvalidator/MailClient/Item
-data/soapvalidator/MailClient/Mail/message_action.xml
```
Comment (prashantajari) [inline conf/ListOfTestCases.txt]: data/soapvalidator/MailClient/Mail/message_action.xml - why is this test skipped?

Comment (girishbhamare) [reply conf/ListOfTestCases.txt]: This has been done as per Previous PR suggestions like "I don't think we need to change legacy edition of this file." by Swapnil. 

So i have deleted all skip files of Zimbra X specially STAF related and copied from Develop branch.

**Comment by swapnilpingle89** [comment]
Please add API team as reviewers as discussed earlier.

File: conf/global.properties (hunk 6 above)
Comment (girishbhamare) [reply conf/global.properties]: done

**Pull request merged**